### PR TITLE
Pie: Add ScrollableVotingEmbed + minor tweak

### PR DESCRIPTION
**Pull request by ArcasCZ**
- moved lock button to middle (between left and right) - it looks overall better
- added `VotableEmbed` extending `nextcord.Embed` and providing callbacks for vote up, neutral and down
- added `ScrollableVotingEmbed` with 3 buttons (vote up, neutral and down) which extends `ScrollableEmbed`

Image of ScrollableVotingEmbed (without scrolling): 
image

### CODE REVIEW

File: pie/utils/objects.py [hunk 1]
```diff
@@ -63,6 +57,14 @@ def __init__(
...
 
+        self.add_item(
+            nextcord.ui.Button(
+                label="\u25b7",
+                style=nextcord.ButtonStyle.green,
+                custom_id="right-button",
+            )
+        )
+
     def __repr__(self):
```
Comment (sinus-x) [inline pie/utils/objects.py:68]: > it looks overall better

Does it though? You are mixing two functionalities together -- scrolling and locking. I think having it behind these two had its reasons. I get that it's nice and symmetric now, but I disagree.

File: pie/utils/objects.py [hunk 2]
```diff
@@ -294,3 +296,109 @@ async def on_timeout(self) -> None:
         """Gets called when the view timeouts."""
         self.value = None
         self.stop()
+
+
+class VotableEmbed(nextcord.Embed):
+    """
+    Abrstract class extendind Embed functionality
```
Comment (sinus-x) [inline pie/utils/objects.py:303]: A typo

File: pie/utils/objects.py [hunk 3]
```diff
@@ -294,3 +296,109 @@ async def on_timeout(self) -> None:
         """Gets called when the view timeouts."""
         self.value = None
         self.stop()
+
+
+class VotableEmbed(nextcord.Embed):
```
Comment (sinus-x) [inline pie/utils/objects.py:301]: Since it must be subclassed, why not add `metaclass=ABCMeta`?

https://docs.python.org/3/library/abc.html

**Review by sinus-x — CHANGES_REQUESTED** [review]
What's the usecase of these two objects? Can't they just be included in the module.py file where they are needed, if they are only used once?

The logic is alright, I'm just worried about extending our utils by not-that-much-useful stuff that may be difficult to remove in the future if it proves to be overkill.

**Comment by ArcasCZ** [comment]
This functionality could be used for any voting. If you pass only one embed, it's still votable. i think we.could reuse this in future.

File: pie/utils/objects.py (hunk 1 above)
Comment (ArcasCZ) [reply pie/utils/objects.py:68]: Well, by the opinion of people from server that were talking about this it does. Having it behind those two has IMHO  no special reason.

File: pie/utils/objects.py (hunk 3 above)
Comment (ArcasCZ) [reply pie/utils/objects.py:301]: Looks promising, I'll try.

File: pie/utils/objects.py (hunk 2 above)
Comment (ArcasCZ) [reply pie/utils/objects.py:303]: Fixed

File: pie/utils/objects.py (hunk 3 above)
Comment (ArcasCZ) [reply pie/utils/objects.py]: Done, working.

File: pie/utils/objects.py (hunk 1 above)
Comment (ArcasCZ) [reply pie/utils/objects.py:69]: OK, so - they are not two completly different functionalities, they work together. So IMHO it's not that big deal to have it inbetween. Also, it looks kinda awful when you have under this other buttons that are in some way symetric, such as Voting buttons.

Comment (ArcasCZ) [reply pie/utils/objects.py:69]: image

Comment (sinus-x) [reply pie/utils/objects.py:69]: Alright.

**Review by sinus-x — APPROVED** [review]
ACK

**Pull request merged**